src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs

```diff
@@ -83,10 +84,28 @@ public async Task SslStream_SendReceiveOverNetworkStream_Ok()
             listener.Stop();
         }
 
-        [Fact]
+        [ConditionalFact]
         [PlatformSpecific(TestPlatforms.Linux)] // This only applies where OpenSsl is used.
         public async Task SslStream_SendReceiveOverNetworkStream_AuthenticationException()
         {
+            SslProtocols clientProtocol;
+            SslProtocols serverProtocol;
+
+            if (PlatformDetection.SupportsTls12 && (PlatformDetection.SupportsTls10 || PlatformDetection.SupportsTls11))
```

Comment:
What's this logic doing? A comment here would be good,

Reply:
added some note. The goal is to cause protocol mismatch. The old test had two values hardcoded.
Instead of making this conditional make skipped on new distributions, I added one more branch as new versions do support 12 and 13. I could possibly rewrite to compare bits and and find any two different values.